apps/emqx_bridge_kafka/src/emqx_bridge_kafka.erl

```diff
@@ -708,3 +719,17 @@ kafka_ext_header_value_validator(Value) ->
                 "placeholder like ${foo}, or a simple string."
             }
     end.
+
+-include_lib("emqx_bridge/include/emqx_bridge_v2_register.hrl").
+
+register_bridge_v2() ->
+    emqx_bridge_v2_register_bridge_type(#{
+        %% Should be provided by all bridges. Even if the bridge_v2_type_name is
+        %% the same as the bridge_v1_type_named.
+        'bridge_v1_type_name' => kafka,
```

Comment:
What if we have new bridges that never had a v1?

Reply:
Then they will still have to have a v1 type name for the compatibility layer. It can be the same as the v2 type name...

Reply:
We could of coarse have a nicer interface that have this assumption and don't force the user to provide both when they are the same...

Reply:
> New bridges after this refactoring might never have had a v1 equivalent. For example, Confluent (which is a shim over Kafka like AEH).

See comments above. Even if they where created after the introduction of v2 they would still need a v1 type for the compatibility layer (even if they are the same). The alternative is to hide certain bridges from the compatibility layer but I don't think that will give us much benefits.

Reply:
I think new bridges shouldn't have a v1 equivalent, since we're moving towards actions and bridges v1 is now a legacy API.  :thinking:

Reply:
The V1 name is now optional.